drivers/include/rn2xx3.h

```diff
@@ -0,0 +1,699 @@
+ * @brief       High-level driver for the RN2483/RN2903 LoRa modules
+ * @{
+ *
+ * @file
+ * @brief       High-level driver for the RN2483/RN2903 LoRa modules
+ *
+ * @author      Alexandre Abadie <alexandre.abadie@inria.fr>
+ */
+
+#ifndef RN2XX3_H
+#define RN2XX3_H
+
+#include <stdint.h>
+
+#include "xtimer.h"
+#include "periph/uart.h"
+#include "periph/gpio.h"
+#include "net/netdev.h"
+
+#include "net/loramac.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief   Maximum length of an exchanged messages with commands
+ */
+#define RN2XX3_MAX_BUF                  (40U)
+
+/**
+ * @brief   Maximum length of an RX message
+ *
+ *          LoRaMAC payload can be up to 250 bytes.
+ */
+#define RN2XX3_RX_MAX_BUF               (250U)
+
+/**
+ * @brief   Maximum delay in second to receive a reply from server.
+ */
+#define RN2XX3_REPLY_DELAY_TIMEOUT      (60U)
+
+/**
+ * @brief   Minimum sleep duration (in ms)
+ */
+#define RN2XX3_SLEEP_MIN                (100U)
+
+/**
+ * @brief   Default sleep duration (in ms)
+ */
+#ifndef RN2XX3_DEFAULT_SLEEP
+#define RN2XX3_DEFAULT_SLEEP            (5000U)
+#endif
+
+#if defined(MODULE_RN2903)
+#define RN2XX3_FREQ_BAND                (915U)
+#endif
+
+#if defined(MODULE_RN2483)
+#define RN2XX3_FREQ_BAND                (868U)
+#endif
+
+/**
+ * @brief   Command responses and server replies status codes
+ */
+enum {
+    RN2XX3_OK                         = 0,   /**< Command is valid */
+    RN2XX3_DATA                       = 1,   /**< Command returned data */
+    RN2XX3_ERR_MAC_INIT               = -1,  /**< Device mac initialization failed */
+    RN2XX3_TIMEOUT                    = -2,  /**< Command timeout */
+    RN2XX3_ERR_INVALID_PARAM          = -3,  /**< Wrong command given */
+    RN2XX3_ERR_NOT_JOINED             = -4,  /**< Network is not joined */
+    RN2XX3_ERR_NO_FREE_CH             = -5,  /**< All channels are busy */
+    RN2XX3_ERR_SILENT                 = -6,  /**< Device is in Silent Immediately state */
+    RN2XX3_ERR_FR_CNT_REJOIN_NEEDED   = -7,  /**< Frame counter rolled over */
+    RN2XX3_ERR_BUSY                   = -8,  /**< MAC is not in Idle state */
+    RN2XX3_ERR_MAC_PAUSED             = -9,  /**< MAC was paused */
+    RN2XX3_ERR_INVALID_DATA_LENGTH    = -10, /**< Wrong payload given */
+    RN2XX3_ERR_KEYS_NOT_INIT          = -11, /**< Keys not configured ("mac join" command) */
+    RN2XX3_REPLY_TX_MAC_OK            = 2,   /**< MAC transmission successful */
+    RN2XX3_REPLY_TX_MAC_ERR           = -12, /**< MAC transmission failed */
+    RN2XX3_REPLY_TX_INVALID_DATA_LEN  = -13, /**< Application payload too large */
+    RN2XX3_REPLY_TX_MAC_RX            = 3,   /**< Data received from server */
+    RN2XX3_REPLY_JOIN_ACCEPTED        = 4,   /**< Join procedure successful */
+    RN2XX3_REPLY_JOIN_DENIED          = -14, /**< Join procedure failed */
+    RN2XX3_REPLY_TIMEOUT              = -15, /**< No MAC reply received from server */
```

Comment:
why is this reply negative and other positiv, looks like this should be an error instead?